code/modules/antagonists/eldritch_cult/eldritch_items.dm

```diff
@@ -315,35 +315,61 @@
+/obj/item/melee/rune_carver/examine(mob/user)
 	. = ..()
-	. += "This item can carve 'Alert carving' - nearly invisible rune that when stepped on gives you a prompt about where someone stood on it and who it was, doesn't get destroyed by being stepped on."
-	. += "This item can carve 'Grasping carving' - when stepped on it causes heavy damage to the legs and stuns for 5 seconds."
-	. += "This item can carve 'Mad carving' - when stepped on it causes dizzyness, jiterryness, temporary blindness, confusion , stuttering and slurring."
+	
+	if((IS_HERETIC(user) || IS_HERETIC_MONSTER(user)))
+		. += "This item can carve 'Alert carving' - nearly invisible rune that when stepped on gives you a prompt about where someone stood on it and who it was, doesn't get destroyed by being stepped on."
+		. += "This item can carve 'Grasping carving' - when stepped on it causes heavy damage to the legs and stuns for 5 seconds."
+		. += "This item can carve 'Mad carving' - when stepped on it causes dizzyness, jiterryness, temporary blindness, confusion , stuttering and slurring."
 
-/obj/item/melee/rune_knife/Initialize()
+/obj/item/melee/rune_carver/Initialize()
 	. = ..()
 	linked_action = new(src)
 
-/obj/item/melee/rune_knife/Destroy()
+/obj/item/melee/rune_carver/Destroy()
 	. = ..()
 	QDEL_NULL(linked_action)
 
-/obj/item/melee/rune_knife/pickup(mob/user)
+/obj/item/melee/rune_carver/pickup(mob/user)
 	. = ..()
-	linked_action.Grant(user, src)
+	if((IS_HERETIC(user) || IS_HERETIC_MONSTER(user)))
+		linked_action.Grant(user, src)
 
-/obj/item/melee/rune_knife/dropped(mob/user, silent)
+/obj/item/melee/rune_carver/dropped(mob/user, silent)
 	. = ..()
 	linked_action.Remove(user, src)
 
-/obj/item/melee/rune_knife/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+/obj/item/melee/rune_carver/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
 	. = ..()
-	if(!is_type_in_typecache(target,blacklisted_turfs) && !drawing && proximity_flag)
+	if(isturf(target) && !is_type_in_typecache(target,blacklisted_turfs) && !drawing && proximity_flag)
 		carve_rune(target,user,proximity_flag,click_parameters)
 
 ///Action of carving runes, gives you the ability to click on floor and choose a rune of your need.
-/obj/item/melee/rune_knife/proc/carve_rune(atom/target, mob/user, proximity_flag, click_parameters)
+/obj/item/melee/rune_carver/proc/carve_rune(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!isturf(target))
+		to_chat(user,"<span class='warning'>The knife doesn't seem capable of 'piercing' anything but the floor...</span>")
+	
+	if(!(IS_HERETIC(user) || IS_HERETIC_MONSTER(user)) && isliving(user))
+		var/mob/living/moron = user
+		to_chat(moron,"<span class='warning'>You lower the knife towards the floor...</span>")
+		if(!do_after(moron,5 SECONDS,target = target))
+			to_chat(moron,"<span class='warning'>You were so close, just need more carving...</span>")
+			return
+		
+		if(!HAS_TRAIT(moron, TRAIT_NAIVE))
```

Comment:
Invert this. `if...else` should not start with a negative condition, it makes it more confusing to read.

Reply:
Noted, I'll fix that up when I move it to it's own proc.